src/sentry/static/sentry/app/views/events/searchBar.tsx

```diff
@@ -25,25 +26,38 @@ const FIELD_TAGS = Object.fromEntries(
   Object.keys(FIELDS).map(item => [item, {key: item, name: item}])
 );
 
-class SearchBar extends React.PureComponent {
-  static propTypes = {
-    api: PropTypes.object,
+type SearchBarProps = {
+  api: Client;
+  organization: SentryTypes.Organization;
+  tags: SentryTypes.Tag;
+  omitTags?: string[];
+  projectIds?: string[];
+  fields?: Field[];
+};
+
+class SearchBar extends React.PureComponent<SearchBarProps> {
+  static propTypes: any = {
+    api: PropTypes.object.isRequired,
     organization: SentryTypes.Organization,
     tags: PropTypes.objectOf(SentryTypes.Tag),
-    omitTags: PropTypes.arrayOf(PropTypes.string),
-    projectIds: PropTypes.arrayOf(PropTypes.number),
-    fields: PropTypes.arrayOf(PropTypes.object),
+    omitTags: PropTypes.arrayOf(PropTypes.string.isRequired),
+    projectIds: PropTypes.arrayOf(PropTypes.number.isRequired),
+    fields: PropTypes.arrayOf(PropTypes.object.isRequired) as any,
   };
 
   componentDidMount() {
     // Clear memoized data on mount to make tests more consistent.
-    this.getEventFieldValues.cache.clear();
+    if (this.getEventFieldValues.cache.clear) {
+      this.getEventFieldValues.cache.clear();
```

Comment:
I would make use of optional chaining

```suggestion
      this.getEventFieldValues.cache?.clear();
```

Reply:
I was going to but it's not `cache` which I'm guarding against but `clear` as it's an optional function on MapCache afaik. Is there a way to avoid an if guard in that case?

Reply:
You can use optional chaining for that too

```tsx
this.getEventFieldValues.cache.clear?.();
```

Reply:
TIL